### ISSUE TITLE
BlockingIOError: [Errno 35] write could not complete without blocking

### ISSUE THREAD

**Issue by jquast** [xonsh/xonsh#2102]
```
+------------------+---------------------+
| xonsh            | 0.5.2               |
| Git SHA          | f1f19f7             |
| Commit Date      | Jan 4 17:47:31 2017 |
| Python           | 3.6.0               |
| PLY              | 3.9                 |
| have readline    | True                |
| prompt toolkit   | None                |
| shell type       | readline            |
| pygments         | None                |
| on posix         | True                |
| on linux         | False               |
| on darwin        | True                |
| on windows       | False               |
| on cygwin        | False               |
| is superuser     | False               |
| default encoding | utf-8               |
| xonsh encoding   | utf-8               |
| encoding errors  | surrogateescape     |
+------------------+---------------------+
```

To cause a `BlockingIOError` exception, I telnet to host `nethack.alt.org`, login, and play a game. On the first game screen/dungeon display, this exception occurs most times, but not every time. Full exception:

```
Exception in thread Thread-10: Traceback (most recent call last):
File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/threading.py", line 916, in _bootstrap_inner
            self.run()
File "/usr/local/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 9834, in run
                i = self._read_write(procout, stdout, sys.__stdout__)
File "/usr/local/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 9877, in _read_write
                                                                       stdbuf.flush()
BlockingIOError: [Errno 35] write could not complete without blocking
```

I tried to make a tty recording, but I can't properly do that because xonsh crashes, and I think there is some more serious problem with tty settings propagating across pseudo terminals.

**Comment by jquast** [comment]
More easily, I simply `telnet nethack.alt.org`, and press `w` to watch games, happens on first screen.

```
 ## nethack.alt.org - http://nethack.alt.org/

 The following games are in progress:

    Username         Game    Size    Start date & time    Idle     W Extra
 a) AbsoluteXero     NH360   91x 26  2017-01-19 23:02:23           0
 b) Amadrulon        NH360  151x 35  2017-01-19 23:12:33           0
 c) BlueWolf         NH343   80x 24  2017-01-19 20:50:53           0   M8
 d) Camtheman88888   NH360  151x 36  2017-01-19 22:32:39  43m 7s   0
 e) Crowther         NH360   80x 24  2017-01-19 22:14:41  11m 53s  0
 f) csundbom2        NH360   80x 24  2017-01-19 21:59:26           0
 g) DlException in thread Thread-4:)
                                   Traceback (most recent call last):
  q) Quit                                                              File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/threading.py", line 916, in _bootstrap_inner
                  self.run()
  =>                          File "/usr/local/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 9834, in run
                      i = self._read_write(procout, stdout, sys.__stdout__)
                                                                             File "/usr/local/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 9877, in _read_write
                                                                             stdbuf.flush()
                                                                                           BlockingIOError: [Errno 35] write could not complete without blocking
```

**Comment by astronouth7303** [comment]
A missed unthreadable program. Just needs to be added to `default_threadable_predictors()`.

There's a line to do it from `.xonshrc`, but I don't know it off-hand.

**Comment by scopatz** [comment]
Thanks for reporting @jquast. I have added a potential fix in #2102

**Comment by scopatz** [comment]
Please feel free to reopen if this continues.
